arbitrum-docs/arbos/l1-to-l2-messaging.mdx

```diff
@@ -1,21 +1,125 @@
+  </NodeDescriptions>
+</MermaidWithHtml>
+
+Without bidirectional cross-layer messaging, it would be impossible to build dApps that interact with both layers, which would severely limit the value and reach of your dApps.
+
+:::caution Editor note
+
+ - Low-confidence intro - this could be improved through more concrete / compelling examples. A "For example..." could help identify relatable situations in which these capabilities become critical. This could help make the information "sticky" for readers who are trying to solve specific problems in addition to learning how the stuff works.
+
+:::
+
+
+## 2: The conceptual primitives
+
+The rest of this concept document focuses on constructing an understanding of L1-to-L2 messaging through the following **conceptual primitives**:
+
+ 1. **Messages**: Pieces of information that are sent from one contract to another, and/or one layer to another.
+ 2. **Retryable tickets**: A type of message sent from L1 chains to L2 chains in order to trigger the execution of an L2-hosted smart contract. Think of these as special-purpose DTOs (data transfer objects) that encapsulate the information needed to execute a smart contract: the contract's address, the function to call, the function's arguments, etc.
+ 3. **Receipts**: todo
+ 4. **ETH deposits**: todo
+ 5. **Delayed Inbox**: todo
+
+
+## 3: Elaborating on the conceptual primitives
+
+Let's explore how each of these primitives relate to one another in the context of cross-layer communication.
+
+
+### 3.1: Messages
+
+ - A **message** is a data structure that contains information about a transaction that was initiated on L1, but that needs to be executed on L2. Messages are sent from L1 to L2, and they're used to trigger contract calls on L2.
+ - This primitive is important to understand if you're building a dApp that... <!-- todo -->
+
+
+:::caution Editor note
+
+ - Unsure about whether or not these things apply only to L1-to-L2 messages, or if they apply to L2-to-L1 messages as well
+ - We could evaluate consolidating cross-layer messaging into this single concept doc
+   - This would allow us to replace the `Cross-chain messaging` node with a single `Cross-chain messaging` document, reducing the number of sidebar clicks required to get to the content
+
+:::
+
+
+### 3.2: Retryable tickets
+
+ - Retryable tickets (retryables) are the canonical data structure that Arbitrum uses to facilitate L1-to-L2 messages. 
+ - Using our cupcake vending machine example, a retryable could be used to solve the following problem: The cupcake contract was deployed to Arbitrum L2, but Alice is an L1 user who isn't comfortable with L2. She wants to give a cupcake to Bob, who is an L2 user. Retryable tickets allow Alice to initiate the transaction on L1, without knowing about L2, and then have the transaction executed on L2.
```

Comment:
Instead of framing around user's comfort levels with L1 vs. L2, I would frame it as: the user has to perform an operation that _requires_ interacting with both an L1 and an L2 contract. Maybe, for example, we describe a "cupcake bridge": Alice has a cupcake on L1 and wants to give it to Bob, who is on L2.